src/math.jl

```diff
@@ -145,7 +145,7 @@ for f in (
     :log, :log2, :log10, :log1p, :exp, :exp2, :exp10, :expm1, :frexp, :exponent,
 )
     @eval function Base.$f(q::UnionAbstractQuantity)
-        iszero(dimension(q)) || throw(DimensionError(q))
+        dimension(q) == dimension(NoDims()) || throw(DimensionError(q))
         return $f(ustrip(q))
     end
 end
```

Comment:
Note that later on you should have an explicit change defining `iszero(::NoDims) = true`, etc.